File: go/fiber/internal/modules/address/service.go
package address

import (
	"tcc/internal/common/errors"
	"tcc/internal/common/messages"
	"tcc/internal/modules/address/dto"
	"tcc/internal/modules/address/entity"
	"tcc/internal/modules/client"
)

type AddressService struct{
	repository struct{
		client client.ClientRepository
		address AddressRepository
	}
}

func (s *AddressService) Create(data dto.CreateAddressDto, idClient string) (*entity.AddressEntity, *errors.BaseError) {
	client, _ := s.repository.client.FindById(idClient)
	if client == nil {
		return nil, errors.BadRequestError(messages.ErrorMessages.Client.NotFound)
	}

	existsAddress, _ := s.repository.address.FindByIdClient(idClient)

	if existsAddress != nil {
		return nil, errors.BadRequestError(messages.ErrorMessages.Address.BadRequest.AlreadyExists)
	}

	newAddress, _ := s.repository.address.Create(ICreateAddressData{
		Logradouro: data.Logradouro,
		Numero: data.Numero,
		Bairro: data.Bairro,
		Cidade: data.Cidade,
		UF: data.Uf,
		CEP: data.Cep,
		IDCliente: idClient,
		Complemento: data.Complemento,
	})

	if newAddress == nil {
		return nil, errors.InternalServerError(messages.ErrorMessages.InternalServer)
	}

	return newAddress, nil
}

File: rust/axum/src/modules/address/service.rs
use super::repository::{AddressRepository, ICreateAddressData};
use super::entity::AddressEntity;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::errors::base::BaseError;
use crate::common::messages::error::ErrorMessages;
use uuid::Uuid;
use crate::modules::client::repository::ClientRepository;
pub struct AddressService;

impl AddressService {
    pub async fn create(
        data: ICreateAddressData,
        id_client: Uuid,
    ) -> Result<AddressEntity, BaseError> {
        let client = ClientRepository::find_by_id(&id_client.to_string()).await.unwrap();

        if client.is_none() {
            return Err(bad_request_error(ErrorMessages::CLIENT_NOT_FOUND, None));
        }

        let exists_address = AddressRepository::find_by_id_client(&id_client.to_string()).await.unwrap();

        if exists_address.is_some() {
            return Err(bad_request_error(ErrorMessages::ADDRESS_BAD_REQUEST_ALREADY_EXISTS, None));
        }

        AddressRepository::create(data)
            .await
            .map_err(|_| bad_request_error(ErrorMessages::INTERNAL_SERVER, None))
    }
}